cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -375,6 +375,14 @@ function(ADD_ARROW_LIB LIB_NAME)
                           LINK_PRIVATE
                           ${ARG_SHARED_PRIVATE_LINK_LIBS})
 
+    if(USE_OBJLIB)
```

Comment:
Can you add a comment explaining this snippet?

Reply:
Added something in c2380ee8a. @kou, please verify if what I wrote for it is correct and complete.

Reply:
Right.